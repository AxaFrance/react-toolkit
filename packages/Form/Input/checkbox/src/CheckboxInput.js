import React from 'react';
import PropTypes from 'prop-types';
import {
  FieldConstants as Constants,
  Field,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import CheckBoxModes from './CheckboxModes';
import Checkbox from './Checkbox';

const propTypes = {
  ...Constants.propTypes,
  mode: PropTypes.oneOf([
    CheckBoxModes.default,
    CheckBoxModes.classic,
    CheckBoxModes.inline,
    CheckBoxModes.toggle,
  ]),
  options: PropTypes.array.isRequired,
  values: PropTypes.array,
};

const defaultProps = {
  ...Constants.defaultProps,
  mode: CheckBoxModes.default,
  values: null,
  className: '',
};

const CheckboxInput = props => {
  const {
    mode,
    messageType,
    message,
    classModifier,
    id,
    name,
    options,
    values,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    helpMessage,
    children,
    placeholder,
    isVisible,
    className,
    onChange,
    readOnly,
    disabled,
    forceDisplayMessage,
    ...otherProps
  } = props;
  let rowModifier = classModifier;
  if (mode === CheckBoxModes.classic) {
    rowModifier += ' label-top';
  }
  const newOptions = InputManager.getOptionsWithId(options);
  const firstId = InputManager.getFirstId(newOptions);
  return (
    <Field
      label={label}
      name={name}
      id={firstId}
      helpMessage={helpMessage}
      message={message}
      childrenRight={children}
      messageType={messageType}
      placeholder={placeholder}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={rowModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Checkbox
        mode={mode}
        name={name}
        id={id}
        isVisible={isVisible}
        values={values}
        options={newOptions}
        onChange={onChange}
        readOnly={readOnly}
        disabled={disabled}
        classModifier={classModifier}
        {...otherProps}
      />
    </Field>
  );
};

CheckboxInput.propTypes = propTypes;
CheckboxInput.defaultProps = defaultProps;

export default CheckboxInput;
