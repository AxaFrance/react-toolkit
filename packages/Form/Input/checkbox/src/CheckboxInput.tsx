import React, { ComponentProps } from 'react';
import { Field } from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import CheckBoxModes from './CheckboxModes';
import Checkbox from './Checkbox';

type Props = ComponentProps<typeof Checkbox> & ComponentProps<typeof Field>;

const CheckboxInput = ({
  mode,
  messageType,
  message,
  classModifier,
  options,
  classNameContainerLabel,
  classNameContainerInput,
  label,
  children,
  placeholder,
  isVisible,
  className,
  forceDisplayMessage,
  ...checkboxProps
}: Props) => {
  const rowModifier = `${classModifier} ${
    mode === CheckBoxModes.classic ? 'label-top' : ''
  }`;
  const newOptions = InputManager.getOptionsWithId(options);

  return (
    <Field
      label={label}
      id={newOptions[0]?.id}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={rowModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Checkbox mode={mode} options={newOptions} {...checkboxProps} />
    </Field>
  );
};

export default CheckboxInput;
