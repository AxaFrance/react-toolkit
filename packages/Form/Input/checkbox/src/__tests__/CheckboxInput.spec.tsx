import React from 'react';
import { render } from '@testing-library/react';
import CheckboxInput from '../CheckboxInput';

describe('<DateInput>', () => {
  it('renders DateInput correctly', () => {
    const { asFragment } = render(
      <CheckboxInput
        label="Image *"
        id="iddateinput"
        name="placeImage"
        options={[]}
        values={[]}
        onChange={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
