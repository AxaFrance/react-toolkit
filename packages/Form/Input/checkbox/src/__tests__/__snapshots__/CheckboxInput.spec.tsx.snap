// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInput> renders DateInput correctly 1`] = `
<DocumentFragment>
  <div
    class="row af-form__group af-form__group--undefined"
  >
    <div
      class="col-md-2"
    >
      <label
        class="af-form__group-label"
      >
        Image *
      </label>
    </div>
    <div
      class="col-md-10"
    />
  </div>
</DocumentFragment>
`;
